charts/prometheus-adapter/templates/deployment.yaml

```
@@ -117,7 +117,9 @@ spec:
         {{- toYaml .Values.affinity | nindent 8 }}
       topologySpreadConstraints:
         {{- toYaml .Values.topologySpreadConstraints | nindent 8 }}
+      {{- if .Values.priorityClassName }}
       priorityClassName: {{ .Values.priorityClassName }}
```

Don't repeat yourself
```suggestion
      {{- with .Values.priorityClassName }}
      priorityClassName: {{ . }}
```

---

Done.